docs/server/source/appendices/all-in-one-bigchaindb.md

```diff
@@ -0,0 +1,84 @@
+# Run BigchainDB with all-in-one Docker
+
+**NOT for Production Use**
```

Comment:
What are the production use criteria? We do not have such warnings in the docs of other setups.

Reply:
I think we all acknowledge that having all services running in a single container is an anti-pattern. This `all-in-one` container is just a convenience and we should have that warning here if not anywhere else.

Reply:
`anti-pattern` is not really an explanation.

Moreover, any setup we have might be labeled "not for production" use at least for a reason that we do not have a stable release yet.

Therefore, to avoid disorienting the users I would refrain from such labels. Instead, we can add a recommendation to use a multi-image setup with a brief explanation on how it is better.

Reply:
> `anti-pattern` is not really an explanation.

Why?

Reply:
`deployment anti-pattern`? I have updated the docs, maybe that is more generic?

Reply:
> > anti-pattern is not really an explanation.

> Why?

`anti-pattern` is just a label.. Does such a setup perform worse? If yes, why? Is it more difficult to maintain such a setup? My point is, we need to describe why we recommend the multi-image setup.

On the other hand, I simply have doubts as to whether this approach has significant drawbacks. So I am asking to prove me wrong. We aim for our components to have a look and feel of a single program and might actually package them into a single binary in the near future - when we do it, there would be simply no choice but to have a single image.

Reply:
Can you comment on the updated docs changes? I kept it more generic now. Also, anti-pattern implies, the suggested micro-service architecture that Docker introduces.

Reply:
The doc changes are fine, thank you! Now the discussion is about removing the "production use" label. Docker is just a tool, the architecture should be determined by the use-cases.

Reply:
> `anti-pattern` is just a label.. Does such a setup perform worse? If yes, why? Is it more difficult to maintain such a setup?

I have no experience regarding this. There are some [sources which suggest](https://blog.couchbase.com/docker-container-anti-patterns/) this is an anti-pattern but the docker documentation has [no mention](https://docs.docker.com/config/containers/multi-service_container/) of it.

I am not really sure if we can say anything about the production readiness of either of these approaches until we try them ourselves?

Reply:
If someone wants to use just a single node in production then maybe yes, because in terms of performance or maintenance it is not a big overhead. If you want to have a network it will be more complex.

Maybe if we have dynamically adding validators implemented, i.e. you deploy one node with this and dynamically add using `upsert`, there are some questions yet to be answered. That is why I suggested not to be used in production.

Reply:
Ah, makes sense!

In other words, a single-node cluster configuration is hardcoded in the image. It is not currently well suited for deploying a network. Could you, perhaps, add such a clarification?